src/core/utils/EventEmitter.ts

```diff
@@ -1,363 +1,15 @@
-import { isArray, isFunction, isObject, isString } from './is.js';
+import * as Ably from 'ably';
 
-function callListener<T, K extends keyof T>(eventThis: { event: K }, listener: EventListener<T, K>, arg: T[K]) {
-  try {
-    listener.apply(eventThis, [arg]);
-  } catch (e) {
-    console.error(
-      'EventEmitter.emit()',
-      'Unexpected listener exception: ' + e + '; stack = ' + (e && (e as Error).stack),
-    );
-  }
-}
-
-/**
- * Remove listeners that match listener
- * @param targetListeners is an array of listener arrays or event objects with arrays of listeners
- * @param listener the listener callback to remove
- * @param eventFilter (optional) event name instructing the function to only remove listeners for the specified event
- */
-export function removeListener<T>(
-  targetListeners: (Function[] | Record<keyof T, Function[]>)[],
-  listener: Function,
-  eventFilter?: keyof T,
-) {
-  let listeners: Function[] | Record<keyof T, Function[]> | undefined;
-  let index: number;
-  let eventName: keyof T;
-
-  for (let targetListenersIndex = 0; targetListenersIndex < targetListeners.length; targetListenersIndex++) {
-    listeners = targetListeners[targetListenersIndex];
-
-    if (isString(eventFilter) && isObject(listeners)) {
-      listeners = listeners[eventFilter];
-    }
-
-    if (isArray(listeners)) {
-      while ((index = listeners.indexOf(listener)) !== -1) {
-        listeners.splice(index, 1);
-      }
-      /* If events object has an event name key with no listeners then
-				 remove the key to stop the list growing indefinitely */
-      const parentCollection = targetListeners[targetListenersIndex];
-      if (eventFilter && listeners.length === 0 && isObject(parentCollection)) {
-        delete parentCollection[eventFilter];
-      }
-    } else if (isObject(listeners)) {
-      for (eventName in listeners) {
-        if (Object.prototype.hasOwnProperty.call(listeners, eventName) && isArray(listeners[eventName])) {
-          removeListener([listeners], listener, eventName);
-        }
-      }
-    }
-  }
-}
-
-// Equivalent of Platform.config.inspect from ably-js for browser/RN
-export function inspect(args: unknown): string {
-  return JSON.stringify(args);
-}
-
-export class InvalidArgumentError extends Error {
-  constructor(...args: [string | undefined]) {
-    super(...args);
-  }
-}
+type Callback<EventsMap> = (arg: EventsMap[keyof EventsMap]) => void;
 
-/**
- * Describes the value of `this` inside the body of an {@link EventListener | `EventListener`} when invoked by an instance of {@link EventEmitter | `EventEmitter`} as the result of an event.
- *
- * @typeParam K The name of the event that this object corresponds to.
- */
-export interface EventListenerThis<K> {
-  /**
-   * The name of the event.
-   */
-  event: K;
+interface InterfaceEventEmitter<EventsMap> extends Ably.EventEmitter<Callback<EventsMap>, void, keyof EventsMap> {
```

Comment:
It would be good if we could typedoc everything in this file, to explain why they're needed etc.